### PR TITLE
ref(ui): Switch to subgrid for gridEditable

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This makes the css much easier to reason about how this works.

There should be no visual changes here.

### CODE REVIEW

File: static/app/components/gridEditable/styles.tsx [hunk 1]
```diff
@@ -95,24 +95,27 @@ export const Grid = styled('table')<{height?: string | number; scrollable?: bool
...
+  grid-column: 1/-1;
+
+  background-color: ${p => p.theme.backgroundSecondary};
+  border-bottom: 1px solid ${p => p.theme.border};
+  font-size: ${p => p.theme.fontSizeSmall};
+  font-weight: 600;
+  line-height: 1;
+  text-transform: uppercase;
+  user-select: none;
+  color: ${p => p.theme.subText};
```
Comment (evanpurkhiser) [inline static/app/components/gridEditable/styles.tsx:115]: We can now apply styles to this component, before styles with `display:content` did nothing.

File: static/app/components/gridEditable/styles.tsx [hunk 2]
```diff
@@ -127,13 +130,6 @@ export const GridHeadCell = styled('th')<{isFirst: boolean; sticky?: boolean}>`
 
   border-right: 1px solid transparent;
   border-left: 1px solid transparent;
-  background-color: ${p => p.theme.backgroundSecondary};
-  color: ${p => p.theme.subText};
-
-  font-size: ${p => p.theme.fontSizeSmall};
-  font-weight: 600;
-  text-transform: uppercase;
-  user-select: none;
```
Comment (evanpurkhiser) [inline static/app/components/gridEditable/styles.tsx:136]: Each cell no longer needs this

File: static/app/components/gridEditable/styles.tsx [hunk 3]
```diff
@@ -95,24 +95,27 @@ export const Grid = styled('table')<{height?: string | number; scrollable?: bool
...
 /**
  * GridHead is the collection of elements that builds the header section of the
  * Grid. As the entirety of the add/remove/resize actions are performed on the
  * header, most of the elements behave different for each stage.
  */
 export const GridHead = styled('thead')`
-  display: contents;
+  display: grid;
+  grid-template-columns: subgrid;
+  grid-column: 1/-1;
```
Comment (evanpurkhiser) [inline static/app/components/gridEditable/styles.tsx:106]: This is the magic. This element now takes up the full row, and is a grid itself where it's columns are inherited from the parent (using `subgrid`)

Comment (JonasBa) [reply static/app/components/gridEditable/styles.tsx:105]: That is so good, it means we can have proper hover states for entire rows right?

**Review by JonasBa — APPROVED** [review]

**Pull request merged**